abel/tools/circularize.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+#########################################################################
+# circularize.py
+# 
+# Image circularization by following peak intensity vs angle
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+# and https://github.com/PyAbel/PyAbel/pull/195
+#
+# Steve Gibson and Dan Hickstein - ideas/code 
+# Jason Gascooke - ideas
+#
+# February 2017
+#########################################################################
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      dr=0.5, dt=0.5, smooth=0, ref_angle=None, 
+                      inverse=False, return_correction=False):
+    """
+    Corrects image distortion on the basis that the structure (Newton spheres)
+    should be circular.  
+
+    This function is especially useful for correcting the image obtained with
+    a velocity-map-imaging spectrometer with imperfect electro-static lenses,
+    allowing a high-resolution 1D photoelectron distribution to be extracted.
+
+    The algorithm splits the image into "slices" at many different angles
+    (set by `dt`) and compares the radial intensity profile of adjacent slices.
+    A scaling factor is found which aligns each slice profile with the previous
+    slice. The image is then corrected using a spline function that smoothly 
+    connects the discrete scaling factors as a continuous function of angle.
+
+    This circularization algorithm should only be applied to a well-centered
+    image, otherwise use the `center` keyword (described below) to center it.
+
```

Comment:
Neat. Thanks for this info. At the moment, something is not quite right with the code auto doc generation. I will check further tomorrow.